### PR TITLE
Working on Hibernate Spacial, fixes #84

### LINKED ISSUE #84: Implement the  r-tree
Implement the r-tree to be scalable. Pair with @JWGmeligMeyling

### PR DESCRIPTION
**Pull request by JWGmeligMeyling**
There is still more work to be done on getting the repositories and build plugins right

### CODE REVIEW

File: src/main/java/me/moodcat/api/models/SongModel.java [hunk 1]
```diff
@@ -66,6 +66,12 @@
      */
     private String artworkUrl;
 
+    @Deprecated
+    private double valence;
+
+    @Deprecated
+    private double arousal;
```
Comment (JeremybellEU) [inline src/main/java/me/moodcat/api/models/SongModel.java]: Deprecated =]

File: src/main/resources/persistence.properties [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+hibernate.dialect = org.hibernate.spatial.dialect.postgis.PostgisDialect
 hibernate.connection.driver_class = org.postgresql.Driver
 hibernate.archive.autodetection = class
 
-javax.persistence.jdbc.url = jdbc:postgresql://188.166.92.29:5432/moodcat_db
-javax.persistence.jdbc.user = moodcat
-#hibernate.hbm2ddl.auto = create-drop
-hibernate.hbm2ddl.auto = update
+javax.persistence.jdbc.url = jdbc:postgresql://localhost:5432/moodcat
+javax.persistence.jdbc.user = postgres
```
Comment (JeremybellEU) [inline src/main/resources/persistence.properties]: Now we can't connect anymore to the main database, so we have to change them back to the original properties.

File: src/test/resources/bootstrap/fall-out-boy.json [hunk 3]
```diff
@@ -10,7 +10,35 @@
 					"artworkUrl": "https://i1.sndcdn.com/artworks-000057992801-cxjxj7-large.jpg",
 					"soundCloudId": 202330997,
 					"duration": 300000
-				}
+				},
+			  {
+				"id": 2,
+				"name": "Thanks for the Memories",
```
Comment (JeremybellEU) [inline src/test/resources/bootstrap/fall-out-boy.json]: We might want to change the data here so we can actually have meaningful songs when running locally.

Comment (JWGmeligMeyling) [inline src/test/resources/bootstrap/fall-out-boy.json]: Yes but this is not related to this PR ;)

File: src/main/resources/persistence.properties (hunk 2 above)
Comment (JWGmeligMeyling) [inline src/main/resources/persistence.properties]: Correct, but  I think it´s best too keep it this way for a while so we can test under postgrse locally using a create-drop schema. And then with deployment switch back to the original settings, when the server is prepared with both the new geometry types and the new user enitity.

File: src/main/java/me/moodcat/api/models/SongModel.java (hunk 1 above)
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/models/SongModel.java]: We might want to leave this in for a while for debugging? So we can see if the sorting works on larger datasets?

**Comment by Jaapp-** [comment]
It needs to be done.
J-just.. just do it...

**Comment by Azkle** [comment]
LGTM
